packages/main/src/models/conversation.ts

```diff
@@ -109,6 +109,7 @@ export function getPresentableElements({
         isRead: cache
           .getFlags(latestEdit ? latestEdit.revision.message.id : message.id)
           .includes("\\Seen"),
+        isStarred: cache.getFlags(message.id).includes("\\Flagged"),
```

Comment:
You need to check whether any revision includes a `"\\Flagged"` flag, not just the first message. At this point in the code every element in the `resources` collection has a `revision` property, which has a reference to the message that provided that revision.

I'm sorry about the weird overlap of the words "revision" and "resource"! For elements in the `resources` collection the `resource` property is a reference to the first revision which gives subsequent revisions a consistent way to identify what they have edited, and the `revision` property is a reference to the edit message that produced the new revision.